baseimage/Makefile

```diff
@@ -1,18 +1,37 @@
 NAME="hyperledger/fabric-baseimage"
+VERSION=0.0.9
+ARCH=$(shell uname -m)
+DOCKER_NAME ?= hyperledger/fabric-baseimage
+DOCKER_TAG ?= $(ARCH)-$(VERSION)
 VAGRANTIMAGE=packer_virtualbox-iso_virtualbox.box
 
+DOCKER_BASE_x86_64=ubuntu:trusty
+
+DOCKER_BASE=$(DOCKER_BASE_$(ARCH))
+
+ifeq ($(DOCKER_BASE), )
+$(error "Architecture \"$(ARCH)\" is unsupported")
+endif
+
+
 # strips off the post-processors that try to upload artifacts to the cloud
 packer-local.json: packer.json
 	jq 'del(."post-processors"[0][1]) | del(."post-processors"[1][1])' packer.json > $@
 
-all:
-	packer build packer.json
+all: vagrant docker
 
 $(VAGRANTIMAGE): packer-local.json
+	BASEIMAGE_RELEASE=$(VERSION) \
 	packer build -only virtualbox-iso packer-local.json
 
-docker: packer-local.json
-	packer build -only docker packer-local.json
+Dockerfile:
+
+docker: Dockerfile.in Makefile
+	@echo "Generating docker"
+	@cat Dockerfile.in | \
```

Comment:
I should emit a "Do not edit!" warning into the resulting Dockerfile, or switch to piping the file in rather than writing it to the filesystem.